dplyr wont compile on R3.4

* installing *source* package 'dplyr' ...
** package 'dplyr' successfully unpacked and MD5 sums checked
** libs
g++ -m64  -I/usr/include/R -DNDEBUG -I../inst/include -DCOMPILING_DPLYR -DBOOST_NO_INT64_T -DBOOST_NO_INTEGRAL_INT64_T -DBOOST_NO_LONG_LONG -DRCPP_USING_UTF8_ERROR_STRING -I"/usr/lib64/R/library/Rcpp/include" -I"/usr/lib64/R/library/BH/include" -I"/usr/lib64/R/library/bindrcpp/include" -I"/usr/lib64/R/library/plogr/include" -I/usr/local/include   -fpic  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic  -c RcppExports.cpp -o RcppExports.o
In file included from ../inst/include/tools/Quosure.h:4,
                 from ../inst/include/tools/tools.h:9,
                 from ../inst/include/dplyr.h:6,
                 from RcppExports.cpp:4:
../inst/include/tools/SymbolString.h: In constructor 'dplyr::SymbolString::SymbolString(const Rcpp::Symbol&)':
../inst/include/tools/SymbolString.h:21: error: no matching function for call to 'Rcpp::String::String(const char*, cetype_t)'
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:159: note: candidates are: Rcpp::String::String(Rbyte)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:158: note:                 Rcpp::String::String(Rcomplex)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:157: note:                 Rcpp::String::String(bool)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:156: note:                 Rcpp::String::String(double)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:155: note:                 Rcpp::String::String(int)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:149: note:                 Rcpp::String::String(const wchar_t*)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:144: note:                 Rcpp::String::String(const char*)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:138: note:                 Rcpp::String::String(const std::wstring&)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:133: note:                 Rcpp::String::String(const std::string&)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:126: note:                 Rcpp::String::String(const Rcpp::String::const_StringProxy&, const std::string&)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:121: note:                 Rcpp::String::String(const Rcpp::String::const_StringProxy&)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:114: note:                 Rcpp::String::String(const Rcpp::String::StringProxy&, const std::string&)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:109: note:                 Rcpp::String::String(const Rcpp::String::StringProxy&)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:91: note:                 Rcpp::String::String(SEXPREC*, const std::string&)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:74: note:                 Rcpp::String::String(SEXPREC*)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:67: note:                 Rcpp::String::String(const Rcpp::String&, const std::string&)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:62: note:                 Rcpp::String::String(const Rcpp::String&)
/usr/lib64/R/library/Rcpp/include/Rcpp/String.h:56: note:                 Rcpp::String::String()
RcppExports.cpp: In function 'SEXPREC* dplyr_get_date_classes_try()':
RcppExports.cpp:267: warning: unused variable 'rcpp_output_type'
RcppExports.cpp:267: warning: unused variable 'rcpp_output_condition'
RcppExports.cpp: In function 'SEXPREC* dplyr_get_time_classes_try()':
RcppExports.cpp:296: warning: unused variable 'rcpp_output_type'
RcppExports.cpp:296: warning: unused variable 'rcpp_output_condition'
RcppExports.cpp: In function 'SEXPREC* dplyr_build_index_cpp_try(SEXPREC*)':
RcppExports.cpp:325: warning: unused variable 'rcpp_output_type'
RcppExports.cpp:325: warning: unused variable 'rcpp_output_condition'
make: *** [RcppExports.o] Error 1
ERROR: compilation failed for package 'dplyr'
* removing '/usr/lib64/R/library/dplyr'

Rcpp is updated to latest version on cran. Have no problem to compile other R packages. When trying to update dplyr to 0.7.1 I get into a loop and the only solution is to ‘force close’. Captured some of the output:
>                                ^
> <scratch space>:72:1: note: expanded from here
> boost_static_assert_typedef_329
> ^
> In file included from arrange.cpp:8:
> In file included from ../inst/include/dplyr/GroupedDataFrame.h:7:
> In file included from ../inst/include/dplyr/Result/GroupedSubset.h:6:
> In file included from ../inst/include/dplyr/DataFrameSubsetVisitors.h:9:
> In file included from ../inst/include/dplyr/subset_visitor.h:4:
> In file included from ../inst/include/dplyr/SubsetVectorVisitor.h:5:
> In file included from ..ccache /usr/local/opt/llvm/bin/clang++ -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -I../inst/include -DCOMPILING_DPLYR -DBOOST_NO_INT64_T -DBOOST_NO_INTEGRAL_INT64_T -DBOOST_NO_LONG_LONG -DRCPP_USING_UTF8_ERROR_STRING -I"/Library/Frameworks/R.framework/Versions/3.4/Resources/library/Rcpp/include" -I"/Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include" -I"/Library/Frameworks/R.framework/Versions/3.4/Resources/library/bindrcpp/include" -I"/Library/Frameworks/R.framework/Versions/3.4/Resources/library/plogr/include" -I/usr/local/include -L/usr/local/opt/llvm/include   -fPIC  -Wall -g -O2  -Wall -g -O2  -O3 -mtune=native -pipe -pedantic -Wall -Wno-unused-command-line-argument  -c between.cpp -o between.o
> /inst/include/dplyr/DataFrameVisitorsIndexMap.h:4:
> In file included from ../inst/include/dplyr/DataFrameVisitors.h:6:
> In file included from ../inst/include/dplyr/visitor_set/VisitorSetMixin.h:5:
> In file included from ../inst/include/dplyr/visitor_set/VisitorSetHash.h:4:
> In file included from ../inst/include/tools/hash.h:10:
> In file included from /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/unordered_map.hpp:17:
> In file included from /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/unordered/unordered_map.hpp:18:
> In file included from /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/unordered/detail/map.hpp:7:
> In file included from /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/unordered/detail/equivalent.hpp:15:
> In file included from /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/unordered/detail/extract_key.hpp:14:
> In file included from /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/unordered/detail/table.hpp:15:
> In file included from /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/unordered/detail/buckets.hpp:16:
> In file included from /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/unordered/detail/allocate.hpp:31:
> In file included from /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/tuple/tuple.hpp:28:
> /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/tuple/detail/tuple_basic.hpp:586:5: warning: unused typedef 'boost_static_assert_typedef_586' [-Wunused-local-typedef]
>     BOOST_STATIC_ASSERT(length<tuple>::value == 2);// check_length = 2
>     ^
> ../inst/include/dplyr/workarounds/static_assert.h:170:16: note: expanded from macro 'BOOST_STATIC_ASSERT'
>                BOOST_JOIN(boost_static_assert_typedef_, __LINE__) BOOST_STATIC_ASSERT_UNUSED_ATTRIBUTE
>                ^
> /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/config/suffix.hpp:556:28: note: expanded from macro 'BOOST_JOIN'
> #define BOOST_JOIN( X, Y ) BOOST_DO_JOIN( X, Y )
>                            ^
> /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/config/suffix.hpp:557:31: note: expanded from macro 'BOOST_DO_JOIN'
> #define BOOST_DO_JOIN( X, Y ) BOOST_DO_JOIN2(X,Y)
>                               ^
> /Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include/boost/config/suffix.hpp:558:32: note: expanded from macro 'BOOST_DO_JOIN2'
> #define BOOST_DO_JOIN2( X, Y ) X##Y
>                                ^
> <scratch space>:74:1: note: expanded from here
> boost_static_assert_typedef_586
> ^
> 3 warnings generated.
> In file included from between.cpp:1:
> In file included from ./pch.h:1:
> In file included from /Library/Frameworks/R.framework/Versions/3.4/Resources/library/Rcpp/include/Rcpp.h:27:
> In file included from /Library/Frameworks/R.framework/Versions/3.4/Resources/library/Rcpp/include/RcppCommon.h:70:
> /Library/Frameworks/R.framework/Versions/3.4/Resources/library/Rcpp/include/Rcpp/utils/tinyformat.h:1046:32: warning: no newline at end of file [-Wnewline-eof]
> #endif // TINYFORMAT_H_INCLUDED
>                                ^
> 1 warning generated.
> ccache /usr/local/opt/llvm/bin/clang++ -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -I../inst/include -DCOMPILING_DPLYR -DBOOST_NO_INT64_T -DBOOST_NO_INTEGRAL_INT64_T -DBOOST_NO_LONG_LONG -DRCPP_USING_UTF8_ERROR_STRING -I"/Library/Frameworks/R.framework/Versions/3.4/Resources/library/Rcpp/include" -I"/Library/Frameworks/R.framework/Versions/3.4/Resources/library/BH/include" -I"/Library/Frameworks/R.framework/Versions/3.4/Resources/library/bindrcpp/include" -I"/Library/Frameworks/R.framework/Versions/3.4/Resources/library/plogr/include" -I/usr/local/include -L/usr/local/opt/llvm/include   -fPIC  -Wall -g -O2  -Wall -g -O2  -O3 -mtune=native -pipe -pedantic -Wall -Wno-unused-command-line-argument  -c bind.cpp -o bind.o




> On 26 Jun 2017, at 11:46, Sergio Oller <notifications@github.com> wrote:
> 
> What version of Rcpp do you have? I would say you need 0.12.9 at least (to have the Rcpp::String::String(const char*, cetype_t) function). dplyr seems to depend on 0.12.6. Could you try again updating Rcpp?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/tidyverse/dplyr/issues/2902#issuecomment-311013166>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADLC2dVqBh86lepqY-oga7mhQ-35NY4uks5sH34LgaJpZM4ODvDX>.
>

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -25,7 +25,7 @@ Imports:
     pkgconfig,
     rlang (>= 0.1),
     R6,
-    Rcpp (>= 0.12.6),
+    Rcpp (>= 0.12.7),
     tibble (>= 1.3.1),
     utils
 Suggests:

